### PR TITLE
Remove `flow` from proposals

### PR DESCRIPTION
**Pull request by jessieay**
* We do not use this concept for serial or parallel steps
* Steps of type `Serial` or `Parallel` are added to a proposal, instead
* Removing at suggestion of https://github.com/18F/C2/pull/835#issuecomment-161836392

### CODE REVIEW

**Comment by jessieay** [comment]
Trello card: https://trello.com/c/0BLjWz1K/56-remove-flow-from-proposals

File: spec/features/ncr/work_orders/requester_edit_spec.rb [hunk 1]
```diff
@@ -141,14 +141,6 @@
...
   end
 
-  scenario "has 'Discard Changes' link" do
-    visit edit_ncr_work_order_path(work_order)
-
-    click_link "Discard Changes"
-
-    expect(current_path).to eq(proposal_path(ncr_proposal))
-  end
-
```
Comment (pkarman) [inline spec/features/ncr/work_orders/requester_edit_spec.rb]: Why is this test deleted?

File: spec/lib/dispatcher_spec.rb [hunk 2]
```diff
@@ -1,9 +1,8 @@
 describe Dispatcher do
   let(:proposal) { create(:proposal) }
 
-  describe '.deliver_new_proposal_emails' do
-    it "uses the LinearDispatcher for linear approvals" do
-      proposal.flow = 'linear'
+  describe ".deliver_new_proposal_emails" do
+    it "uses the LinearDispatcher for non-NCR approvals" do
```
Comment (pkarman) [inline spec/lib/dispatcher_spec.rb]: What happens when we add a client that isn't NCR that does not use LinearDispatcher?

File: app/decorators/proposal_decorator.rb [hunk 3]
```diff
@@ -9,26 +9,8 @@ def total_approvers
...
-        CASE steps.status
-        WHEN 'approved' THEN 1
-        WHEN 'actionable' THEN 2
-        ELSE 3
-        END
-      SQL
-    )
-  end
-
   def steps_in_list_order
```
Comment (pkarman) [inline app/decorators/proposal_decorator.rb]: Even though we are not dropping the `flow` column, a Proposal still has a a "flow" flavor, indicated by the root step type. Would a proposal with a Parallel root step still need `steps_by_status`?

File: app/models/proposal.rb [hunk 4]
```diff
@@ -74,14 +72,6 @@ def root_step
...
   end
 
-  def parallel?
-    flow == "parallel"
-  end
-
-  def linear?
-    flow == "linear"
-  end
-
```
Comment (pkarman) [inline app/models/proposal.rb]: `parallel?` and `linear?` still seem like useful methods to have? Maybe they should just look at the root step instead?

**Comment by pkarman** [comment]
Is it really true that "We do not use this concept for serial or parallel steps" ? Isn't `flow == 'linear'` the same thing as `proposal.individual_steps.first.type == "Steps::Serial"` ?

I guess what I'm saying is that even if we do not store the proposal step flavor as an explicit string in the database, we still have the concept of a proposal having a step flavor, as indicated by its root step. If we are going to keep Parallel step logic around (as I think we should, per conversation in #835), then I think we should keep all the view logic in place, and maybe just create a custom read-only `Proposal.flow` method that checks the root step type instead of a column value.

File: spec/features/ncr/work_orders/requester_edit_spec.rb (hunk 1 above)
Comment (jessieay) [inline spec/features/ncr/work_orders/requester_edit_spec.rb]: hah WOOPS! accident. Was failing so I was editing it but must have accidentally deleted- will restore!

File: spec/lib/dispatcher_spec.rb (hunk 2 above)
Comment (jessieay) [inline spec/lib/dispatcher_spec.rb]: I am not sure. One of the challenges of having the concept of parallel steps without using them is that we don't have test cases on the integration level for the situation where we are using them.

File: app/models/proposal.rb (hunk 4 above)
Comment (jessieay) [inline app/models/proposal.rb]: what if a proposal has some linear steps and some parallel?

File: app/views/proposals/show.html.haml [hunk 5]
```diff
@@ -75,31 +69,6 @@
     %li
       = link_to "View history", history_proposal_path(@proposal)
 
-- # @todo: there are multiple approval-status ids
-- if @proposal.parallel?
```
Comment (jessieay) [inline app/views/proposals/show.html.haml]: this is what a parallel proposal looks like when there are no approvals and when all parallel steps are approved:

screen shot 2015-12-16 at 10 05 21 am

When I update the status on the proposal itself to "approved", this is what it looks like:

screen shot 2015-12-16 at 10 06 56 am

File: app/views/shared/_email_status.html.erb [hunk 6]
```diff
@@ -28,7 +28,7 @@
                   </td>
                   <td>
                     <%- last_approver = index == @proposal.steps_in_list_order.count - 1 %>
-                    <%= status_icon_tag(display_status, @proposal.linear?, last_approver) %>
+                    <%= status_icon_tag(display_status, last_approver) %>
```
Comment (jessieay) [inline app/views/shared/_email_status.html.erb]: this is what the email looks like for parallel approvals before this was removed

screen shot 2015-12-16 at 10 13 12 am

File: app/decorators/proposal_decorator.rb (hunk 3 above)
Comment (jessieay) [inline app/decorators/proposal_decorator.rb]: `steps_in_list_order` is used in the emails. I just created a proposal on master with a parallel flow and 2 steps, this is what the email looks like:

screen shot 2015-12-16 at 10 13 12 am

seems like this method is not working

File: app/models/proposal.rb (hunk 4 above)
Comment (pkarman) [inline app/models/proposal.rb]: root step dictates flavor, imo

**Comment by pkarman** [comment]
agreed, those screenshots really illustrate the problem. I'm fine to drop all the view-related code as part of this PR.

Let's resolve #879 first, then return to this PR since a new `flow` method would depend upon #879 to work.

**Comment by jessieay** [comment]
`serial?` and `parallel?` methods added at e6c1a4591274e24f6a5d65fa481e210d8261faa3

Root step PR is completed.

This is ready for re-review

**Comment by pkarman** [comment]
Perfect, thanks @jessieay. I will QA.

**Comment by pkarman** [comment]
no regressions found in c2-dev

**Pull request merged**